subsys/net/lib/http/http_server_http1.c

```diff
@@ -314,11 +314,35 @@ static int on_header_field(struct http_parser *parser, const char *at,
 						   struct http_client_ctx,
 						   parser);
 
+	if (strncmp(at, "Upgrade", 7) == 0) {
```

Comment:
Such an approach is not 100% robust, HTTP1 parser's `http_data_cb()` does not guarantee to provide complete header field/value in a single call. I've experienced this when testing with very small RX buffer sizes, in case an incomplete header is read with `zsock_recv()`, the parser will call the callback twice (or even more for large fields). This case has been addressed in `on_url()` cb, we should do the same for headers (prob need some transient buffer).

Reply:
Indeed, good point, will fix.